extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwner.kt

```diff
@@ -0,0 +1,192 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import com.mapbox.common.Logger
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.ScreenCoordinate
+import com.mapbox.maps.plugin.animation.camera
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * @see MapboxCarMap to create new map experiences.
+ *
+ * Maintains the surface state for [MapboxCarMap].
+ */
+@OptIn(MapboxExperimental::class)
+internal class CarMapSurfaceOwner {
+
+  internal var mapboxCarMapSurface: MapboxCarMapSurface? = null
+    private set
+  internal var visibleArea: Rect? = null
+    private set
+  internal var edgeInsets: EdgeInsets? = null
+    private set
+  internal var visibleCenter: ScreenCoordinate = visibleCenter()
+    private set
+
+  private val carMapObservers = CopyOnWriteArraySet<MapboxCarMapObserver>()
+
+  fun registerObserver(mapboxCarMapObserver: MapboxCarMapObserver) {
+    carMapObservers.add(mapboxCarMapObserver)
+    Logger.i(TAG, "registerObserver + 1 = ${carMapObservers.size}")
+
+    mapboxCarMapSurface?.let { carMapSurface ->
+      mapboxCarMapObserver.onAttached(carMapSurface)
+    }
+    ifNonNull(mapboxCarMapSurface, visibleArea, edgeInsets) { _, area, edge ->
+      Logger.i(TAG, "registerObserver visibleAreaChanged")
+      mapboxCarMapObserver.onVisibleAreaChanged(area, edge)
+    }
+  }
+
+  fun unregisterObserver(mapboxCarMapObserver: MapboxCarMapObserver) {
+    carMapObservers.remove(mapboxCarMapObserver)
+    mapboxCarMapSurface?.let { mapboxCarMapObserver.onDetached(it) }
+    Logger.i(TAG, "unregisterObserver - 1 = ${carMapObservers.size}")
+  }
+
+  fun clearObservers() {
+    val oldCarMapSurface = this.mapboxCarMapSurface
+    oldCarMapSurface?.let { carMapObservers.forEach { it.onDetached(oldCarMapSurface) } }
+    carMapObservers.clear()
+  }
+
+  fun surfaceAvailable(mapboxCarMapSurface: MapboxCarMapSurface) {
+    Logger.i(TAG, "surfaceAvailable")
+    val oldCarMapSurface = this.mapboxCarMapSurface
+    this.mapboxCarMapSurface = mapboxCarMapSurface
+    oldCarMapSurface?.let { carMapObservers.forEach { it.onDetached(oldCarMapSurface) } }
+    carMapObservers.forEach { it.onAttached(mapboxCarMapSurface) }
```

Comment:
This has a little different behavior. I didn't need this to be a certain way downstream, but it would be good to make a definition that we obeyed in every case for onDetached. You can also access the map surface outside of the callback

``` kotlin
    carMapSurfaceOwner.registerObserver(object : MapboxCarMapObserver {
      override fun onDetached(mapboxCarMapSurface: MapboxCarMapSurface) {
        // carMapSurfaceOwner.mapboxCarMapSurface != mapboxCarMapSurface
      }
    })
```

So if we change it here, we also want to change it here
``` kotlin
fun surfaceDestroyed() {
--snip--
    this.mapboxCarMapSurface = null
    detachSurface?.let { carMapObservers.forEach { it.onDetached(detachSurface) } }
```

Reply:
Added unit tests to capture this, that way the behavior we choose is on purpose. 

it is more meaningful to have access to the updated state. So leaving the state change before the onDetached call